src/ng/compile.js

```diff
@@ -3526,6 +3570,11 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
             break;
 
           case '&':
+            if (!hasOwnProperty.call(attrs, attrName) && !optional && strictComponentBindingsEnabled) {
```

Comment:
Nit: Check the booleans first to avoid unnecessary method calls. E.g.:

```js
if (strictComponentBindingsEnabled && !optional && !hasOwnProperty.call(attrs, attrName)) {...
```